src/test/java/de/tum/in/www1/artemis/migration/MigrationIntegrityTest.java

```diff
@@ -0,0 +1,246 @@
+package de.tum.in.www1.artemis.migration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+import static tech.jhipster.config.JHipsterConstants.SPRING_PROFILE_TEST;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Profiles;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.config.migration.MigrationEntry;
+import de.tum.in.www1.artemis.config.migration.MigrationRegistry;
+import de.tum.in.www1.artemis.domain.MigrationChangelog;
+import de.tum.in.www1.artemis.migration.entries.TestChangeEntry20211214_231800;
+import de.tum.in.www1.artemis.migration.entries.TestChangeEntry20211215_231800;
+import de.tum.in.www1.artemis.migration.entries.TestChangeEntry20211216_231800;
+import de.tum.in.www1.artemis.repository.MigrationChangeRepository;
+
+public class MigrationIntegrityTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    private MigrationRegistry registrySpyMock;
```

Comment:
Is it really necessary to mock and spy here?

Reply:
I refactored it in a way that made it not necessary to mock anymore.

The reason I initially chose to mock it was to do more of a whitebox approach. However, I think the information rate is not too interesting so we can also use the approach without mocking. It makes sense to don't slow the tests down even more.